packages/thirdweb/src/utils/fetch.ts

```diff
@@ -10,18 +10,19 @@ import {
 
 const DEFAULT_REQUEST_TIMEOUT = 60000;
 
-const FETCH_CACHE = new WeakMap<
-  { client: ThirdwebClient; ecosystem?: Ecosystem },
+const FETCH_CACHE = new Map<
+  string,
   (url: string, init?: RequestInit) => Promise<Response>
 >();
 
 /**
  * @internal
  */
 export function getClientFetch(client: ThirdwebClient, ecosystem?: Ecosystem) {
-  if (FETCH_CACHE.has({ client, ecosystem })) {
+  const CACHE_KEY = JSON.stringify({ client, ecosystem });
+  if (FETCH_CACHE.has(CACHE_KEY)) {
     // biome-ignore lint/style/noNonNullAssertion: the `has` above ensures that this will always be set
-    return FETCH_CACHE.get({ client, ecosystem })!;
+    return FETCH_CACHE.get(CACHE_KEY)!;
   }
```

Comment:
we should benchmark that this cache actually makes a difference that is more significant the the time it takes to JSON stringify this on every request

Reply:
It wasn't any faster with caching (about the same) so I removed it